packages/devtools/test/inspector_controller_test.dart

```diff
@@ -544,5 +544,71 @@ void main() async {
 
       await env.tearDownEnvironment();
     });
+
+    test('hotRestart', () async {
+      await env.setupEnvironment();
+
+      // Select row index 4.
+      // The important thing about this is that the details tree should scroll
+      // instead of re-rooting as the selected row is already visible in the
+      // details tree.
+      tree.onTap(const Offset(0, rowHeight * 4.5));
+      expect(
+        tree.toStringDeep(),
+        equalsIgnoringHashCodes(
+          '▼[R] root ]\n'
+              '  ▼[M] MyApp\n'
+              '    ▼[M] MaterialApp\n'
+              '      ▼[S] Scaffold\n'
+              '      ├───▼[C] Center <-- selected\n'
+              '      │     ▼[/icons/inspector/textArea.png] Text\n'
+              '      └─▼[A] AppBar\n'
+              '          ▼[/icons/inspector/textArea.png] Text\n',
+        ),
+      );
+
+      await serviceManager.performHotRestart();
+      // The isolate starts out paused on a hot restart so we have to resume
+      // it manually to make the test pass.
+      await serviceManager.service
```

Comment:
@devoncarew pauseOnResume after restart is probably an edge case we haven't thought through fully yet for general devtools-debugger integration. I'm not sure if there is a way to implement it currently without things getting quite odd if you have multiple debuggers attached.

Reply:
Definitely would require some thought. Either coordination about who's in charge of resuming, or a convention that you only resume if you're the tool that launched the app in the first place.

Reply:
Some other ideas:
1. Eliminate the pause altogether. I'm not clear why we need pause on resume after a hot restart. Shouldn't the VM just maintain the same list of breakpoints that existed before the hot restart? The pause on initial app start case seems like it should only have 1 debugger connected so isn't a problem.
2. Only resume after all VM Service clients that have opted into pause on start behavior have asked to resume. Would require adding an extra vm service method for a client to opt in that it needs to be notified on resumes after hot restart.